addons/sale_stock/models/stock.py

```diff
@@ -123,7 +123,7 @@ def _compute_sale_order_ids(self):
             ]).mapped('move_id')
             stock_moves = stock_moves.search([('id', 'in', stock_moves.ids)]).filtered(
                 lambda move: move.picking_id.location_dest_id.usage == 'customer' and move.state == 'done')
-            lot.sale_order_ids = stock_moves.mapped('sale_line_id.order_id')
+            lot.sale_order_ids = stock_moves.sudo().mapped('sale_line_id.order_id')
```

Comment:
doesn't seems legit, if you don't have the groups `group_sale_salesman_all_leads`, you shouldn't see the fields `sale_order_ids` and  `sale_order_count`. maybe add a group into these fields instead, What do you think ?

Reply:
I don't know. :thinking: 
In this use case, there is a conflict in the rights.
If I have group `stock.group_stock_user`, I can read/write on `sale.order`
https://github.com/odoo/odoo/blob/3c3b4bf9fa711c21ae2142a3260805820a410b6f/addons/sale_stock/security/ir.model.access.csv#L6
So I should normally be able to see these fields.

Reply:
So a fix would be to remove the group 'own documemts only' on the user ?

Reply:
@Whenrow Yep. That would fix the issue.

It is just strange that an user with `Inventory: User` rights and no Sales rights can access `sale_order_ids`, but as soon as we give him `Sales: User: Own Documents Only` rights, he can't anymore.

I can maybe apply `sudo` only if I can write on `sale.order`.
Does it seem like a good idea?
Or is it better to add group `group_sale_salesman_all_leads` to these fields as suggested by @ryv-odoo?

Reply:
The good solution would be to ignore the `Sales: User: Own Documents Only` in case of stock user group for all the other cases that could failed like this one. Maybe change the record rule.
If it's not possible I would choose the sudo solution.

Reply:
Hi @Whenrow 

I couldn't find a good way to cancel or modify the record rule for this special case.
I've handled the `sudo` in a `try...except` instead.

Reply:
@kitan191 Not a huge fan of the try...except. What about this ?
```python
if self.env.user.has_group('stock.group_stock_user'):
    lot.sale_order_ids = stock_moves.sudo().mapped('sale_line_id.order_id')
else:
    lot.sale_order_ids = stock_moves.mapped('sale_line_id.order_id')
```

Reply:
It's good for me. :ok_hand: 
Thanks